### PR TITLE
Window Flashing

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Your game window will now flash orange if alt-tabbed on AdminPM/AdminHelp, being put into DNA scanner for cloning, defibbing, Strange Reagent, and any events (borers, ERT, aliens, blob, etc.)

There is a preference to disable it though

Credit to TGStation
:cl:
add: Your game window will now flash on receiving Admin PMs, being revived, and ghost roles. You can disable this in Game Preferences.
/:cl:

### CODE REVIEW

**Comment by Crazylemon64** [comment]
Ah, neat! :+1:

**Comment by Spacemanspark** [comment]
This is a God send.
:+1:

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -471,3 +472,8 @@ proc/pollCandidates(var/Question, var/be_special_type, var/antag_age_check = 0,
 			candidates.Remove(G)
 
 	return candidates
+
+/proc/window_flash(var/client_or_usr)
```
Comment (Fox-McCloud) [inline code/__HELPERS/game.dm]: no implied vars, please.

**Comment by tigercat2000** [comment]
I think this should probably be a preference option. Otherwise, it can get annoying, especially if you normally auto-hide the taskbar.

**Comment by uraniummeltdown** [comment]
There should now be a pref to disable the feature

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -471,3 +472,11 @@ proc/pollCandidates(var/Question, var/be_special_type, var/antag_age_check = 0,
 			candidates.Remove(G)
 
 	return candidates
+
+/proc/window_flash(client/C)
+    if(ismob(C))
```
Comment (Crazylemon64) [inline code/__HELPERS/game.dm]: If the given mob lacks a client, won't the statement on line 480 runtime, as indexing `C.prefs` will attempt to reference something off of `null`?

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -471,3 +472,12 @@ proc/pollCandidates(var/Question, var/be_special_type, var/antag_age_check = 0,
 			candidates.Remove(G)
 
 	return candidates
+
+/proc/window_flash(client/C)
+	if(ismob(C))
+		var/mob/M = C
+		if(M.client)
+			C = M.client
+	if(!C.prefs.windowflashing || !C)
```
Comment (Crazylemon64) [inline code/__HELPERS/game.dm]: oh, this should be the other way around, or it'll evaluate `C.prefs.windowflashing` when `C` is null and runtime anyways

**Comment by Crazylemon64** [comment]
Sorry for not addressing this for so long, real life has been keeping me down!

**Pull request merged**